module/actor/data/character.ts

```diff
@@ -0,0 +1,1570 @@
+  }
+
+  /* ---------------------------------------- */
+
+  #applyCharacterBonuses() {
+    this.#applyBonusesToAttributes()
+    this.#applyBonusesToHitLocations()
+  }
+
+  /* ---------------------------------------- */
+
+  #applyBonusesToAttributes() {
+    for (const bonus of this._globalBonuses) {
+      if (bonus.type !== 'attribute') continue
+
+      const attrKey = bonus.attrkey
+      if (this.attributes[attrKey as keyof typeof this.attributes]) {
+        this.attributes[attrKey as keyof typeof this.attributes].value += bonus.mod as number
+      }
+
+      if (attrKey === 'DODGE') {
+        // Special handling for dodge attribute
+        this.dodge.value += bonus.mod as number
+      }
+    }
+  }
+
+  /* ---------------------------------------- */
+
+  #applyBonusesToHitLocations() {
+    for (const location of this.hitlocations) {
+      location.drItem = this._drBonusesFromItems[location.where] ?? 0
+      const newDR = location.import + location.drItem + location.drMod
+      // NOTE: I'm unsure as to whether drCap should ever apply.
+      // On the one hand, using the ! operator in the /dr command may imply
+      // that the DR of the location should always be set to the new value.
+      // On the other hand, this may not be the intended behavior.
+      // This is where documentation regarding intent would be helpful.
+      location._dr = location.drCap === null ? newDR : Math.max(location.drCap, newDR)
+    }
+  }
+
+  /* ---------------------------------------- */
+
+  #prepareLiftingMoving() {
+    const basicLift = 5 * (this.attributes.ST.value * this.attributes.ST.value)
+
+    this.liftingmoving = {
+      basiclift: basicLift,
+      onehandedlift: basicLift * 2,
+      twohandedlift: basicLift * 8,
+      shove: basicLift * 12,
+      runningshove: basicLift * 24,
+      carryonback: basicLift * 15,
+      shiftslightly: basicLift * 50,
+    }
+  }
+
+  /* ---------------------------------------- */
+
+  #prepareEncumbrance() {
+    const automaticEncumbrance =
+      game.settings?.get(Settings.SYSTEM_NAME, Settings.SETTING_AUTOMATIC_ENCUMBRANCE) ?? false
+
+    let manualEncumbranceIndex = -1
+    let foundCurrent = false
+    const currentEncumbrance = this.encumbrance ?? []
+    if (!currentEncumbrance.length && !automaticEncumbrance) {
+      manualEncumbranceIndex = currentEncumbrance.findIndex(enc => enc.current)
+      foundCurrent = true
+    }
+
+    const encumbrance: fields.SchemaField.SourceData<EncumbranceSchema>[] = []
+    const basicLift = this.liftingmoving.basiclift
+    const liftBrackets: number[] = [basicLift, basicLift * 2, basicLift * 3, basicLift * 6, basicLift * 10]
+    const basicMove = this.basicmove.value
+    const basicDodge = this.dodge.value
+
+    const moveIsEnhanced = this.currentMoveMode?.enhanced !== null
+
+    const carriedWeight = this.eqtsummary.eqtlbs ?? 0
+
+    for (let i = 0; i < liftBrackets.length; i++) {
+      let move = Math.max(1, basicMove * (1 - i * 0.2))
```

Comment:
Bug or feature? On a hex grid, drop all fractions. If playing gridless, maybe we want to keep the fraction?